app/services/doc_auth/mock/result_response.rb

```diff
@@ -136,6 +142,10 @@ def doc_auth_result_from_uploaded_file
         parsed_data_from_uploaded_file&.[]('doc_auth_result')
       end
 
+      def portrait_match_results
+        parsed_data_from_uploaded_file.dig('portrait_match_results')&.deep_symbolize_keys
```

Comment:
This will be included the mocked TrueIDResponse, where the generator generate error like following:

https://github.com/18F/identity-idp/blob/0d2d7942546cef573700919756e1cc469567156c/app/services/doc_auth/error_generator.rb#L192-L195

Though the test following generates expected error, it's because the _FaceMatchResult_ field is missing.

https://github.com/18F/identity-idp/blob/0d2d7942546cef573700919756e1cc469567156c/spec/services/doc_auth/mock/doc_auth_mock_client_spec.rb#L340-L342

So either we change the error_generator.rb or fix here to make the filed name as expected. I understood that we wanna use _face_match_result_ name in yaml, to achieve that we can change the method like following:

```ruby
def portrait_match_results
    # this will camelize key face_match_result to FaceMatchResult
    parsed_data_from_uploaded_file.dig('portrait_match_results')&.
      transform_keys!{ |key| key.to_s.camelize }.
      deep_symbolize_keys
end
```

At the same time, we can add a positive test case for passed liveness in _doc_auth_mock_client_spec.rb_ to catch this.

Reply:
Re: testing, [does this test from the PR cover](https://github.com/18F/identity-idp/pull/9882/files#diff-29b763e9ae2d7b70e0a21cb59e96d1dee4abf0cb6a56bb5c6b49441f38616818R288) what you mean about positive liveness?

Re: `portrait_match_results`, that's a good point. I would rather leave `error_generator` as is for this work, since it's focused on the mock and the generator will change implementation. I thought about using the camelized `FaceMatchResult` as the yml key, but it doesn't match our other yml values. I think your workaround is a good way to get past it. This was changed in [a757b79](https://github.com/18F/identity-idp/pull/9882/commits/a757b79593933b176cb94dacc89c991805eb0c27).